In [1]:
import numpy as np
import pandas as pd

In [2]:
data_gs = pd.read_csv("../data/observations-gregtasney.csv")
data_gs.head()

C:\Users\lisa_\AppData\Local\Temp\ipykernel_7724\3140657315.py:1: DtypeWarning: Columns (15,26,29,30,31) have mixed types. Specify dtype option on import or set low_memory=False.
  data_gs = pd.read_csv("../data/observations-gregtasney.csv")


id                          observed_on_string observed_on  \
0  26102174  Sun May 26 2019 07:56:38 GMT+1000 (GMT+10)  2019-05-26   
1  26102183  Sun May 19 2019 10:45:52 GMT+1000 (GMT+10)  2019-05-19   
2  26102189  Sun May 19 2019 10:37:34 GMT+1000 (GMT+10)  2019-05-19   
3  26102706                                         NaN         NaN   
4  26102759                      2019/05/01 8:04 PM HST  2019-05-01   

          time_observed_at time_zone  user_id  user_login    user_name  \
0  2019-05-25 21:56:38 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
1  2019-05-19 00:45:52 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
2  2019-05-19 00:37:34 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
3                      NaN    Hawaii  1809954  gregtasney  Greg Tasney   
4  2019-05-02 06:04:00 UTC    Hawaii  1809954  gregtasney  Greg Tasney   

                created_at               updated_at  ... place_county_name  \
0  2019-05-31 06:25:48 UTC  2019-06-02 04:59:50 UTC  ...         Salisbury   
1  2019-05-31 06:26:15 UTC  2022-10-18 09:01:06 UTC  ...           Corinda   
2  2019-05-31 06:26:36 UTC  2022-01-25 09:25:37 UTC  ...           Corinda   
3  2019-05-31 06:55:01 UTC  2023-04-27 13:22:35 UTC  ...           Corinda   
4  2019-05-31 06:57:25 UTC  2019-06-02 04:59:40 UTC  ...         Salisbury   

  place_state_name place_country_name place_admin1_name place_admin2_name  \
0       Queensland          Australia        Queensland         Salisbury   
1       Queensland          Australia        Queensland           Corinda   
2       Queensland          Australia        Queensland           Corinda   
3       Queensland          Australia        Queensland           Corinda   
4       Queensland          Australia        Queensland         Salisbury   

     species_guess    scientific_name           common_name  \
0    Argina astrea      Argina astrea  crotalaria pod borer   
1  Lesser Wanderer     Danaus petilia       Lesser Wanderer   
2    Blue Wanderer    Tirumala hamata         Blue Wanderer   
3      Burnie Vine  Malaisia scandens           Burnie Vine   
4    Argina astrea      Argina astrea  crotalaria pod borer   

   iconic_taxon_name  taxon_id  
0            Insecta  318525.0  
1            Insecta  333775.0  
2            Insecta  197078.0  
3            Plantae  705655.0  
4            Insecta  318525.0  

[5 rows x 42 columns]

In [4]:
data_gs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 97617 entries, 0 to 97616
Data columns (total 42 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   id                                97617 non-null  int64  
 1   observed_on_string                97614 non-null  object 
 2   observed_on                       97614 non-null  object 
 3   time_observed_at                  97594 non-null  object 
 4   time_zone                         97617 non-null  object 
 5   user_id                           97617 non-null  int64  
 6   user_login                        97617 non-null  object 
 7   user_name                         97617 non-null  object 
 8   created_at                        97617 non-null  object 
 9   updated_at                        97617 non-null  object 
 10  quality_grade                     97617 non-null  object 
 11  license                           97617 non-null  object 
 12  url 

In [65]:
data_gs.iloc[-6:-1, 9:23]

updated_at quality_grade   license  \
97611  2023-12-03 11:56:25 UTC      research  CC-BY-SA   
97612  2023-12-04 03:02:00 UTC      research  CC-BY-SA   
97613  2023-12-04 08:24:19 UTC      research  CC-BY-SA   
97614  2023-12-04 05:07:50 UTC      needs_id  CC-BY-SA   
97615  2023-12-04 08:25:27 UTC      research  CC-BY-SA   

                                                     url  \
97611  https://www.inaturalist.org/observations/19285...   
97612  https://www.inaturalist.org/observations/19285...   
97613  https://www.inaturalist.org/observations/19285...   
97614  https://www.inaturalist.org/observations/19285...   
97615  https://www.inaturalist.org/observations/19285...   

                                               image_url  \
97611                                                NaN   
97612                                                NaN   
97613                                                NaN   
97614  https://inaturalist-open-data.s3.amazonaws.com...   
97615                                                NaN   

                                               sound_url tag_list description  \
97611  https://static.inaturalist.org/sounds/854912.m...      NaN         NaN   
97612  https://static.inaturalist.org/sounds/854918.m...      NaN         NaN   
97613  https://static.inaturalist.org/sounds/854919.m...      NaN         NaN   
97614                                                NaN      NaN         NaN   
97615  https://static.inaturalist.org/sounds/854922.m...      NaN         NaN   

       num_identification_agreements  num_identification_disagreements  \
97611                              1                                 0   
97612                              2                                 0   
97613                              2                                 0   
97614                              0                                 0   
97615                              2                                 0   

       captive_cultivated  oauth_application_id  \
97611               False                   2.0   
97612               False                   2.0   
97613               False                   2.0   
97614               False                   2.0   
97615               False                   2.0   

                           place_guess   latitude  
97611  Barney View QLD 4287, Australia -28.272760  
97612  Barney View QLD 4287, Australia -28.272459  
97613  Barney View QLD 4287, Australia -28.272466  
97614  Barney View QLD 4287, Australia -28.272574  
97615  Barney View QLD 4287, Australia -28.272747

In [61]:
# suspect user_id is same for all entries, using value_counts to confirm
data_gs["user_id"].value_counts()

1809954    97617
Name: user_id, dtype: int64

In [62]:
# suspect user_login is same for all entries, using value_counts to confirm
data_gs["user_login"].value_counts()

gregtasney    97617
Name: user_login, dtype: int64

In [63]:
# suspect user_name is same for all entries, using value_counts to confirm
data_gs["user_name"].value_counts()

Greg Tasney    97617
Name: user_name, dtype: int64

In [66]:
# suspect license is all the same, confirmed
data_gs["license"].value_counts()

CC-BY-SA    97617
Name: license, dtype: int64

In [73]:
# suspect license is all the same, confirmed
data_gs["oauth_application_id"].value_counts()

3.0    12091
2.0       81
Name: oauth_application_id, dtype: int64

In [60]:
# list of columns with null values
col_with_null = []
for col in data_gs.columns:
    if data_gs[col].describe()[0] != data_gs.shape[0]:
        col_with_null.append(col)
print(f'No. of columns with null values = {len(col_with_null)}')
col_with_null

No. of columns with null values = 28


['observed_on_string',
 'observed_on',
 'time_observed_at',
 'image_url',
 'sound_url',
 'tag_list',
 'description',
 'oauth_application_id',
 'place_guess',
 'latitude',
 'longitude',
 'positional_accuracy',
 'public_positional_accuracy',
 'geoprivacy',
 'taxon_geoprivacy',
 'positioning_method',
 'positioning_device',
 'place_town_name',
 'place_county_name',
 'place_state_name',
 'place_country_name',
 'place_admin1_name',
 'place_admin2_name',
 'species_guess',
 'scientific_name',
 'common_name',
 'iconic_taxon_name',
 'taxon_id']

In [71]:
# The columns that contribute to only 5% or fewer of the entries
for col in col_with_null:
    rows_notnull = data_gs[data_gs[col].notnull()].shape[0]
    if rows_notnull <= (data_gs.shape[0] / 20):
        print(col, ":", f'   no. of rows not null = {rows_notnull}\n')

sound_url :    no. of rows not null = 144

tag_list :    no. of rows not null = 19

description :    no. of rows not null = 4843

geoprivacy :    no. of rows not null = 462

positioning_method :    no. of rows not null = 73

positioning_device :    no. of rows not null = 73

place_town_name :    no. of rows not null = 421



In [78]:
# The columns that have fewer than 1% with null values
for col in col_with_null:
    rows_null = data_gs[data_gs[col].isnull()].shape[0]
    if rows_null <= (data_gs.shape[0] / 100):
        print(col, ":", f'   no. of null rows= {rows_null}\n')

observed_on_string :    no. of null rows= 3

observed_on :    no. of null rows= 3

time_observed_at :    no. of null rows= 23

image_url :    no. of null rows= 116

place_guess :    no. of null rows= 306

latitude :    no. of null rows= 306

longitude :    no. of null rows= 306

place_county_name :    no. of null rows= 321

place_state_name :    no. of null rows= 306

place_country_name :    no. of null rows= 306

place_admin1_name :    no. of null rows= 306

place_admin2_name :    no. of null rows= 321

scientific_name :    no. of null rows= 1

iconic_taxon_name :    no. of null rows= 2

taxon_id :    no. of null rows= 1



In [67]:
data_gs[data_gs['tag_list'].notnull()].iloc[:, :17]

id       observed_on_string observed_on  \
7381    45669589  2013/01/11 5:28 PM AEDT  2013-01-11   
7382    45669590  2013/01/11 6:23 PM AEDT  2013-01-11   
7383    45669592  2013/01/11 7:24 PM AEDT  2013-01-11   
7384    45669594  2013/01/31 6:45 PM AEDT  2013-01-31   
7385    45669595  2013/02/02 7:36 PM AEDT  2013-02-02   
7386    45669596  2013/02/04 6:24 PM AEDT  2013-02-04   
70422  145417256       2022/12/30 1:09 PM  2022-12-30   
72536  146675416      2023/01/15 10:43 AM  2023-01-15   
76741  149717679       2023/02/25 7:46 AM  2023-02-25   
76820  149719214       2023/02/25 8:21 AM  2023-02-25   
76849  149769473       2023/02/25 8:24 AM  2023-02-25   
77294  150158654       2023/03/03 8:03 PM  2023-03-03   
77746  150380681       2023/03/05 3:36 PM  2023-03-05   
77761  150380703       2023/03/05 4:26 PM  2023-03-05   
77901  150643678       2023/03/08 8:18 PM  2023-03-08   
78071  150909418       2023/03/12 9:26 AM  2023-03-12   
78156  150911935      2023/03/12 10:54 AM  2023-03-12   
78282  151004994      2023/03/12 12:06 PM  2023-03-12   
78291  151021129      2023/03/12 12:15 PM  2023-03-12   

              time_observed_at time_zone  user_id  user_login    user_name  \
7381   2013-01-11 06:28:00 UTC    Sydney  1809954  gregtasney  Greg Tasney   
7382   2013-01-11 07:23:00 UTC    Sydney  1809954  gregtasney  Greg Tasney   
7383   2013-01-11 08:24:00 UTC    Sydney  1809954  gregtasney  Greg Tasney   
7384   2013-01-31 07:45:00 UTC    Sydney  1809954  gregtasney  Greg Tasney   
7385   2013-02-02 08:36:00 UTC    Sydney  1809954  gregtasney  Greg Tasney   
7386   2013-02-04 07:24:00 UTC    Sydney  1809954  gregtasney  Greg Tasney   
70422  2022-12-30 03:09:00 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
72536  2023-01-15 00:43:00 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
76741  2023-02-24 20:46:00 UTC    Sydney  1809954  gregtasney  Greg Tasney   
76820  2023-02-24 21:21:00 UTC    Sydney  1809954  gregtasney  Greg Tasney   
76849  2023-02-24 21:24:00 UTC    Sydney  1809954  gregtasney  Greg Tasney   
77294  2023-03-03 10:03:00 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
77746  2023-03-05 05:36:00 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
77761  2023-03-05 06:26:00 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
77901  2023-03-08 10:18:00 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
78071  2023-03-11 23:26:00 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
78156  2023-03-12 00:54:00 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
78282  2023-03-12 02:06:00 UTC  Brisbane  1809954  gregtasney  Greg Tasney   
78291  2023-03-12 02:15:00 UTC  Brisbane  1809954  gregtasney  Greg Tasney   

                    created_at               updated_at quality_grade  \
7381   2020-05-12 10:52:29 UTC  2020-05-12 12:37:07 UTC      research   
7382   2020-05-12 10:52:30 UTC  2020-06-16 05:34:37 UTC      research   
7383   2020-05-12 10:52:31 UTC  2020-05-12 12:36:55 UTC      research   
7384   2020-05-12 10:52:32 UTC  2022-06-28 06:50:26 UTC      research   
7385   2020-05-12 10:52:33 UTC  2020-06-16 05:35:42 UTC      research   
7386   2020-05-12 10:52:35 UTC  2020-05-12 22:36:25 UTC      research   
70422  2022-12-30 11:01:22 UTC  2022-12-30 20:28:09 UTC      research   
72536  2023-01-16 07:50:42 UTC  2023-07-22 23:07:14 UTC      needs_id   
76741  2023-02-26 11:47:01 UTC  2023-02-26 12:23:36 UTC      research   
76820  2023-02-26 12:18:06 UTC  2023-05-12 13:16:21 UTC      research   
76849  2023-02-26 22:11:42 UTC  2023-04-29 12:20:49 UTC      research   
77294  2023-03-03 21:53:01 UTC  2023-03-03 21:53:01 UTC      needs_id   
77746  2023-03-06 03:50:10 UTC  2023-03-06 03:50:10 UTC      needs_id   
77761  2023-03-06 03:50:15 UTC  2023-03-09 23:51:06 UTC      needs_id   
77901  2023-03-09 10:12:23 UTC  2023-03-09 10:12:23 UTC      needs_id   
78071  2023-03-12 09:54:16 UTC  2023-03-12 09:54:16 UTC      needs_id   
78156  2023-03-12 10:36:58 UTC  2023-03-12 10:36:58 UTC      needs_id  

In [68]:
data_gs[data_gs['description'].notnull()]['description']

17                                    Planted early 2000's
104                                            On Myoporum
113                Escaped or dumped pet. Pet trade morph.
171      No records on iNaturlist. It's a common plant ...
265      I prefer Oplismenus aemulus which iNaturlist h...
                               ...                        
97484                                  On Smilax australis
97487        Pandorea sp. (Mt Maroon P.I.Forster+ PIF7111)
97517                          N.punctata or N.longifolia.
97545                                   A strange outlier.
97562    Terrible pics. Just couldn't get the short len...
Name: description, Length: 4843, dtype: object

In [70]:
# percent of entries with data from this column
data_gs[data_gs['description'].notnull()].shape[0] / data_gs.shape[0]

0.04961226015960335

In [88]:
data_gs[data_gs['latitude'].isnull()].iloc[:, 21:37]

place_guess  latitude  longitude  positional_accuracy  \
6789          NaN       NaN        NaN                 10.0   
11292         NaN       NaN        NaN                  NaN   
11373         NaN       NaN        NaN               1118.0   
11380         NaN       NaN        NaN                 65.0   
11389         NaN       NaN        NaN               1055.0   
...           ...       ...        ...                  ...   
88333         NaN       NaN        NaN                  NaN   
88334         NaN       NaN        NaN                  NaN   
88335         NaN       NaN        NaN                  NaN   
88336         NaN       NaN        NaN                  NaN   
91850         NaN       NaN        NaN                  NaN   

       public_positional_accuracy geoprivacy taxon_geoprivacy  \
6789                      29727.0    private         obscured   
11292                     29680.0    private         obscured   
11373                     29680.0    private         obscured   
11380                     29680.0    private         obscured   
11389                     29680.0    private         obscured   
...                           ...        ...              ...   
88333                     29680.0    private              NaN   
88334                     29680.0    private             open   
88335                     29680.0    private              NaN   
88336                     29680.0    private              NaN   
91850                         NaN        NaN              NaN   

       coordinates_obscured positioning_method positioning_device  \
6789                   True                NaN                NaN   
11292                  True                NaN                NaN   
11373                  True                NaN                NaN   
11380                  True                NaN                NaN   
11389                  True                NaN                NaN   
...                     ...                ...                ...   
88333                  True                NaN                NaN   
88334                  True                NaN                NaN   
88335                  True                NaN                NaN   
88336                  True                NaN                NaN   
91850                 False                NaN                NaN   

      place_town_name place_county_name place_state_name place_country_name  \
6789              NaN               NaN              NaN                NaN   
11292             NaN               NaN              NaN                NaN   
11373             NaN               NaN              NaN                NaN   
11380             NaN               NaN              NaN                NaN   
11389             NaN               NaN              NaN                NaN   
...               ...               ...              ...                ...   
88333             NaN               NaN              NaN                NaN   
88334             NaN               NaN              NaN                NaN   
88335             NaN               NaN              NaN                NaN   
88336             NaN               NaN              NaN                NaN   
91850             NaN               NaN              NaN                NaN   

      place_admin1_name place_admin2_name  
6789                NaN               NaN  
11292               NaN               NaN  
11373               NaN               NaN  
11380               NaN               NaN  
11389               NaN               NaN  
...                 ...               ...  
88333               NaN               NaN  
88334               NaN               NaN  
88335               NaN               NaN  
88336               NaN               NaN  
91850               NaN               NaN  

[306 rows x 16 columns]

In [80]:
#Geoprivacy can be converted to dummies IF removal of missing lat and lon leaves anything other than NaN
data_gs["geoprivacy"].value_counts()

private     305
obscured    157
Name: geoprivacy, dtype: int64

In [81]:
#taxon_geoprivacy can be converted to dummies IF removal of missing lat and lon leaves anything other than open
data_gs["taxon_geoprivacy"].value_counts()

open        31363
obscured     1652
Name: taxon_geoprivacy, dtype: int64

In [87]:
data_gs[data_gs['latitude'].notnull()].iloc[:, 21:37]

place_guess   latitude   longitude  \
0         Russ Hall Park, Salisbury, QLD, AU -27.550703  153.034180   
1         87 Archer Parade, Corinda, QLD, AU -27.548263  152.994475   
2      430 Cliveden Avenue, Corinda, QLD, AU -27.550458  152.993713   
3                    Brisbane QLD, Australia -27.550470  152.994234   
4                    Brisbane QLD, Australia -27.550649  153.034376   
...                                      ...        ...         ...   
97612        Barney View QLD 4287, Australia -28.272459  152.769096   
97613        Barney View QLD 4287, Australia -28.272466  152.769034   
97614        Barney View QLD 4287, Australia -28.272574  152.769949   
97615        Barney View QLD 4287, Australia -28.272747  152.769823   
97616      Burnett Creek QLD 4310, Australia -28.244983  152.551623   

       positional_accuracy  public_positional_accuracy geoprivacy  \
0                      5.0                         5.0        NaN   
1                      5.0                         5.0        NaN   
2                      5.0                         5.0        NaN   
3                     15.0                        15.0        NaN   
4                      NaN                         NaN        NaN   
...                    ...                         ...        ...   
97612                 77.0                        77.0        NaN   
97613                 80.0                        80.0        NaN   
97614                 35.0                        35.0        NaN   
97615                 49.0                        49.0        NaN   
97616                 31.0                        31.0        NaN   

      taxon_geoprivacy  coordinates_obscured positioning_method  \
0                  NaN                 False                NaN   
1                  NaN                 False                NaN   
2                  NaN                 False                NaN   
3                  NaN                 False                NaN   
4                  NaN                 False                NaN   
...                ...                   ...                ...   
97612             open                 False                gps   
97613             open                 False                gps   
97614              NaN                 False                gps   
97615             open                 False                NaN   
97616              NaN                 False                NaN   

      positioning_device place_town_name   place_county_name place_state_name  \
0                    NaN             NaN           Salisbury       Queensland   
1                    NaN             NaN             Corinda       Queensland   
2                    NaN             NaN             Corinda       Queensland   
3                    NaN             NaN             Corinda       Queensland   
4                    NaN             NaN           Salisbury       Queensland   
...                  ...             ...                 ...              ...   
97612                gps             NaN  Beaudesert  - Pt C       Queensland   
97613                gps             NaN  Beaudesert  - Pt C       Queensland   
97614                gps             NaN  Beaudesert  - Pt C       Queensland   
97615                NaN             NaN  Beaudesert  - Pt C       Queensland   
97616                NaN             NaN              Boonah       Queensland   

      place_country_name place_admin1_name   place_admin2_name  
0              Australia        Queensland           Salisbury  
1              Australia        Queensland             Corinda  
2              Australia        Queensland             Corinda  
3              Australia        Queensland             Corinda  
4              Australia        Queensland           Salisbury  
...                  ...               ...                 ...  
97612          Australia        Queensland  Beaudesert  - Pt C  
97613          Australia        Queensland  Beaudeser

In [84]:
data_gs['place_county_name'].value_counts()

Salisbury                                  5493
Guanaba-Springbrook                        4845
Brookfield (incl. Brisbane Forest Park)    4756
Caloundra - Rail Corridor                  4558
Boonah                                     4142
                                           ... 
San Francisco                                 1
Dakabin-Kallangur-M. Downs                    1
Spring Hill                                   1
Jamboree Heights                              1
Launceston - Pt B                             1
Name: place_county_name, Length: 267, dtype: int64

In [90]:
data_gs['place_admin2_name'].value_counts()

Salisbury                                  5493
Guanaba-Springbrook                        4845
Brookfield (incl. Brisbane Forest Park)    4756
Caloundra - Rail Corridor                  4558
Boonah                                     4142
                                           ... 
San Francisco                                 1
Dakabin-Kallangur-M. Downs                    1
Spring Hill                                   1
Jamboree Heights                              1
Launceston - Pt B                             1
Name: place_admin2_name, Length: 267, dtype: int64

In [85]:
data_gs['place_state_name'].value_counts()

Queensland                      91450
New South Wales                  2986
Western Australia                2368
Australian Capital Territory      499
Tasmania                            7
California                          1
Name: place_state_name, dtype: int64

In [89]:
data_gs['place_admin1_name'].value_counts()

Queensland                      91450
New South Wales                  2986
Western Australia                2368
Australian Capital Territory      499
Tasmania                            7
California                          1
Name: place_admin1_name, dtype: int64

In [86]:
data_gs['place_country_name'].value_counts()

Australia        97310
United States        1
Name: place_country_name, dtype: int64

## Notes on Cleaning:

1. id:  __leave in place__
1. observed_on_string:  __DELETE COLUMN__ (duplicate into of observed_on)
1. observed_on: __leave in place__ (only three null values which will be deleted when time_observed_at is cleaned)
1. time_observed_at: __DELETE 23 NULL ROWS__ (very key information and very small set for deletion)
1. user_id: __DELETE COLUMN__ redundant--simply says all data is Greg's
1. user login: __DELETE COLUMN__ redundant--simply says all data is Greg's
1. user_name: __DELETE COLUMN__ redundant--simply says all data is Greg's
1. created_at: __leave in place__
1. updated_at: __leave in place__
1. quality_grade: __leave in place__
1. license: __DELETE COLUMN__ all entries are identical
1. url: __leave in place for now__ 
1. image_url: __CONVERT TO BINARY__ (if an image_url exists, 1, else 0) 
1. sound_url: __CONVERT TO BINARY__ (if a sound_url exists, 1, else 0) 
1. tag_list: __CONVERT TO BINARY__ (present or not, only 19 entries that have data here, mabye delete later)
1. description: __CONVERT TO BINARY__ (present or not; VERY INTERESTING but not likely useful for current analysis and contributes to less than 5% of the entries)
1. num_identification_agreements: __leave in place__
1. num_identification_disagreements: __leave in place__
1. captive_cultivated: __leave in place__
1. oauth_application_id: __DUMMY VARIABLE__ (only 2.0 or 3.0 or null)
1. place_guess: __leave in place__ (same removal as longitude and latitude--keep in separate data base)
1. latitude: __REMOVE 306 NULL ROWS__ (same removal as longitude and place_guess--keep in separate data base)
1. longitude: __REMOVE 306 NULL ROWS__ (same removal as latitude and place_guess--keep in separate data base)
1. positional_accuracy: 
1. public_positional_accuracy:
1. geoprivacy: __DUMMY__ (private or obscured) IF removal of missing lat, lon leaves anything other than null
1. taxon_geoprivacy: __DUMMY__ (open or obscured) IF removal of missing lat, lon leaves anything other than open
1. coordinates_obscured: __leave in place__ (already binary) UNLESS removal of missing lat, long leaves only 1 response
1. positioning_method:
1. positioning_device:
1. place_town name: __DELETE COLUMN__ only 421 entries, and can be reconstructed from lat/lon
1. place_county_name: __IMPUTE FROM LAT, LON__ (assume that this plus deleting missing lat/lon will cover it)
1. place_state_name: __leave in place__ (assuming removal of 306 missing lat/lon leaves no more nulls here)
1. place_country_name: __leave in place__ (assuming removal of 306 missing lat/lon leaves no more nulls here)
1. place_admin1_name: __DELETE COLUMN__ redundant with place_state_name
1. place_admin2_name: __DELETE COLUMN__ redundant with place_county_name
1. species_guess:
1. scientific_name:
1. common_name:
1. iconic_taxon_name:
1. taxon_id:

## Questions for Stephen/in general

- Should the urls be saved in a different table? The way that SQL organizes dataframes? 
- Should tag list and descriptions be turned to binary (present or not?)
- For research: What is a license? Is it unique to every user?
- how do I double check for sure that place_admin2_name is same as place_county_name?